File: node.js/src/original/persistence/books/bookEntity.ts
import { AuthorEntity } from '../authors';
import { PublisherEntity } from '../publishers/publisherEntity';
import { ReviewerEntity } from '../reviewers';
import { ChapterEntity, FormatEntity } from './entities';
import { CommitteeApprovalVO, TranslationVO } from './valueObjects';

export type BookEntity = {
  id: string;
  currentState: State;
  title: string;
  author: AuthorEntity;
  publisher: PublisherEntity;
  edition: number;
  genre?: string | null;
  isbn?: string | null;
  publicationDate?: Date | null;
  totalPages?: number | null;
  numberOfIllustrations: number | null;
  bindingType?: string | null;
  summary?: string | null;
  committeeApproval?: CommitteeApprovalVO | null;
  reviewers: ReviewerEntity[];
  chapters: ChapterEntity[];
  translations: TranslationVO[];
  formats: FormatEntity[];
};

export enum State {
  Writing = 'Writing',
  Editing = 'Editing',
  Printing = 'Printing',
  Published = 'Published',
  OutOfPrint = 'OutOfPrint',
}

File: csharp/Slimmed/PublishingHouse.Persistence/Books/BookEntity.cs
using PublishingHouse.Persistence.Authors;
using PublishingHouse.Persistence.Books.Entities;
using PublishingHouse.Persistence.Books.ValueObjects;
using PublishingHouse.Persistence.Publishers;
using PublishingHouse.Persistence.Reviewers;

namespace PublishingHouse.Persistence.Books;

public class BookEntity
{
    public enum State { Writing, Editing, Printing, Published, OutOfPrint }

    public required Guid Id { get; set; }
    public required State CurrentState { get; set; }
    public required string Title { get; set; }

    public Guid AuthorId { get; set; }
    public AuthorEntity Author { get; set; } = default!;

    public Guid PublisherId { get; set; }
    public PublisherEntity Publisher { get; set; } = default!;
    public required int Edition { get; set; }
    public string? Genre { get; set; }
    public string? ISBN { get; set; }
    public DateOnly? PublicationDate { get; set; }
    public int? TotalPages { get; set; }
    public int? NumberOfIllustrations { get; set; }
    public string? BindingType { get; set; }
    public string? Summary { get; set; }
    public CommitteeApprovalVO? CommitteeApproval { get; set; }
    public required List<ReviewerEntity> Reviewers { get; set; } = new();
    public required List<ChapterEntity> Chapters { get; set; } = new();
    public required List<TranslationVO> Translations { get; set; } = new();
    public required List<FormatEntity> Formats { get; set; } = new();

    public int Version { get; set; }
}
